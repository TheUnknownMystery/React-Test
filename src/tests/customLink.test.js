import renderer from 'react-test-renderer';
import { LinkDataFixture } from './fixtures/LinkDataFixture';
import CustomLink from '../components/customLink';

test('render react custom link element with props', () => {
  const wrapper = renderer.create(<CustomLink {...LinkDataFixture} />);

  // eslint-disable-next-line testing-library/await-async-query
  const link = wrapper.root.findByProps({ testID: 'custom-link' });

  expect(link.props.href).toBe(LinkDataFixture.link);
  expect(link.children).toEqual([LinkDataFixture.name]);
  expect(wrapper.toJSON()).toMatchSnapshot();
});
