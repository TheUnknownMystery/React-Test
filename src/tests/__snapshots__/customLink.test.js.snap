// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render react custom link element with props 1`] = `
<div>
  <a
    href="https://www.youtube.com/"
    rel="noopener noreferrer"
    target="_blank"
    testID="custom-link"
  >
    this is a test link
  </a>
</div>
`;
